lib/bson/regexp.rb

```diff
@@ -196,15 +195,14 @@ def respond_to?(method, include_private = false)
       #   and 'u' to make \w, \W, etc. match unicode.
       #
       # @param [ BSON::ByteBuffer ] buffer The byte buffer to append to.
-      # @param [ true, false ] validating_keys
       #
       # @return [ BSON::ByteBuffer ] The buffer with the encoded object.
       #
       # @see http://bsonspec.org/#/specification
       #
       # @since 4.2.0
-      def to_bson(buffer = ByteBuffer.new, validating_keys = Config.validating_keys?)
-        return compile.to_bson(buffer, validating_keys) if options.is_a?(Integer)
+      def to_bson(buffer = ByteBuffer.new)
+        return compile.to_bson(buffer) if options.is_a?(Integer)
```

Comment:
Didn't you remove integer options?

Reply:
You're right, let me open a followup ticket.